File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.cpp
class Solution {
public:
    int maxDistance(vector<int>& nums1, vector<int>& nums2) {
        int n=nums1.size(),ans=0;
        for(int i=0;i<n;i++)
        {
           int low=i,high=nums2.size()-1,mid;
            while(low<=high)
            {
                mid=low+(high-low)/2;
                if(nums1[i] <= nums2[mid])
                {
                    ans=max(ans,mid-i);
                    low=mid+1;
                }
                else
                    high=mid-1;
            }
        }
        return ans;
    }
};

File: 1855-maximum-distance-between-a-pair-of-values/1855-maximum-distance-between-a-pair-of-values.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i=0
        j=0
        ans=0
        n1=len(nums1)
        n2=len(nums2)
        while(i<n1 and j<n2):
            if(nums1[i] > nums2[j]):
                i+=1
            else:
                ans=max(ans,j - i)
                j+=1
        return ans